### PR TITLE
Small changes to spot.py to improve readability and have consistent numpy behavior

### PR DESCRIPTION
**Pull request by effting**
Created a separate pull-request for this as requested.

### CODE REVIEW

File: src/odemis/util/spot.py [hunk 1]
```diff
@@ -255,17 +256,19 @@ def MaximaFind(image, qty, len_object=18):
 
     # dilation
     structure = _CreateSEDisk(len_object // 2)
-    dilated = cv2.dilate(filtered, structure, iterations=1)
+    dilated = cv2.dilate(filtered, structure)
 
     # find local maxima
-    binary = (filtered == dilated)
+    binary = (filtered == dilated).astype(numpy.uint8)
```
Comment (pieleric) [inline src/odemis/util/spot.py]: Could you explain what is this needed? Normally it returns an array of booleans. Maybe that is not compatible with cv2? (I guess that booleans are stored as uint8 anyway, so that's why it'd work anyway).

Comment (effting) [reply src/odemis/util/spot.py]: Yes; I think this has to do with cv2. Note that I didn't added the .astype() call; I just moved it up by a few lines. As you say it might indeed just work without but I think it is better to explicitly cast the array to what is expected by cv2.

File: src/odemis/util/spot.py [hunk 2]
```diff
@@ -255,17 +256,19 @@ def MaximaFind(image, qty, len_object=18):
 
     # dilation
     structure = _CreateSEDisk(len_object // 2)
-    dilated = cv2.dilate(filtered, structure, iterations=1)
+    dilated = cv2.dilate(filtered, structure)
```
Comment (tepals) [inline src/odemis/util/spot.py]: Why did you remove this, is `iterations=1` the default?

File: src/odemis/util/spot.py (hunk 1 above)
Comment (tepals) [reply src/odemis/util/spot.py]: Yes, this is because of cv2. It doesn't accept it without the .astype() call

File: src/odemis/util/spot.py (hunk 2 above)
Comment (effting) [reply src/odemis/util/spot.py]: Yes; default is iterations=1; see [link](https://docs.opencv.org/2.4/modules/imgproc/doc/filtering.html#void%20dilate(InputArray%20src,%20OutputArray%20dst,%20InputArray%20kernel,%20Point%20anchor,%20int%20iterations,%20int%20borderType,%20const%20Scalar&%20borderValue))
